### PR TITLE
Parse summary

### PR DESCRIPTION
**Pull request by Redmar-van-den-Berg**
Parse the separate summary output files for SNPs and indels for each sample into two tables, and also add an html graph showing the indel and SNP precision and sensitivity for each sample for easy comparison.

### CODE REVIEW

File: clinical-validation.wdl [hunk 1]
```diff
@@ -216,7 +237,49 @@ workflow ClinicalValidation {
...
+    output {
+        File? IndelTSV = indelTSV
+        File? SnpTSV = snpTSV
+        File? HtmlGraph = htmlGraph
+    }
+
+    parameter_meta {
+        sampleNames: {description:  "The names of the samples, in the same order as snpSummary and indelSummary", category: "common"}
+        dockerImage: {description: "The docker images used.", category: "required"}
+        memory: {description: "The amount of memory to use.", category: "required"}
```
Comment (DavyCats) [inline clinical-validation.wdl]:
```suggestion
        dockerImage: {description: "The docker images used.", category: "advanced"}
        memory: {description: "The amount of memory to use.", category: "advanced"}
```

File: clinical-validation.wdl [hunk 2]
```diff
@@ -216,7 +237,49 @@ workflow ClinicalValidation {
...
+            ~{"--html-graph " + htmlGraph}
+    }
+
+    output {
+        File? IndelTSV = indelTSV
+        File? SnpTSV = snpTSV
+        File? HtmlGraph = htmlGraph
+    }
+
+    parameter_meta {
```
Comment (DavyCats) [inline clinical-validation.wdl:280]: Most inputs are missing here

**Review by DavyCats — DISMISSED** [review]

File: src/parse_summary.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        for line in fin:
+            # Skip divider
+            if line.strip() == '-'*100:
+                continue
+            spline = line.strip().split()
+            data.append({key: value for key, value in zip(header, spline)})
+        return data
+
+
+def print_tsv(data, filename):
```
Comment (rhpvorderman) [inline src/parse_summary.py:41]: This function is a bit complicated due to the writefile partial. why not:
```python
    with open(filename, 'wt') as fout:
        fout.write("Sample\t" + "\t".join(expected_header) + "\n")
        for sample in data:
            for entry in data[sample]:
                fout.write(sample + "\t" + "\t".join(entry[field] for field in expected_header) + "\n")
```

**Review by rhpvorderman — APPROVED** [review]

File: src/parse_summary.py (hunk 3 above)
Comment (rhpvorderman) [reply src/parse_summary.py:41]: ~~Whoops. Forgot the `"\n"` s. But anyway. No need to write a partial function when Python File objects have write methods.~~ Updated the example

**Pull request merged**